### PR TITLE
fix bug with lazyloadingcachepolicy not updating _last_updated on 304

### PR DESCRIPTION
**Pull request by tyroneabdy**
This would happen when a request came in that would hit a not modified
content set from configcat, which would then mean every subsequent
request would also go through the same network request to configcat
to determine that config flag data had not change.

So change it that a 304 or otherwise is counted as updated
so that it reduces load and time.
Also remove the case where if multiple attempts at getting flags from configcat
within the same request would all continue to retrieve the information.

### CODE REVIEW

**Comment by tim-clipchamp** [comment]
As context, we are using the lazy loading cache policy with a 5 minute time to live. Our flag config is updated perhaps 1-2 times per day -- i.e. much longer than the time to live value.

After the 5 minutes is up, without this change, _every_ call to `get_value()` will include attempting to fetch the flags from the CDN (and getting a 304). I can see that each request takes between a couple of hundred ms, and about 1.5s at the tail. However for some of the requests that we serve, there are 10s of flags that we query the state of, resulting in a latency measured in the 10s of seconds, purely from us trying to fetch the state of the flags 10s of times (it also uses up a lot of our quota).

In other words, a 304 should be considered as a successful fetch of the flags, and should reset `_last_updated`.
